Negative test cases for BNP

j1mehta opened the pull request.

hp-networking commented:
**Initial Comments:
baremetal_network_provisioning/tests/api/test_bnp_negative.py
ip_address - variable keep the name which is consistant with other names. -> sw_ip_address
inf configuration - change to hpe_bnp
Add cases to supports,
1. update bnp_switch - with read only attributes
2. update switch - with invalid keys 
3. update switch with invalid credentials
4. make switch ports participated in ironic and try to delete
5. Try to create same switch with same ip address
6 create switch with wrong community
7. Show and delete switch 
8. test invalid vendor
9. invalid protocol


remove 
sw_usr = switch_username
+sw_passwd = switch_password
why are we keeping this ?
baremetal_network_provisioning/tests/tempest/config.py
Line 21 : hpeBNP_OPTS - in python all the variable should be lower case. move to hpe_bnp_opts 
Line 23: ip_address  - > sw_ip_address
please remove the hardcoded ip address 105.0.1.266  use ip_address

in sample file 
access_protocol_v3 not handled?

Important question : are you expecting any real switch should be present in the environment ?
As per the code **

j1mehta commented:
Thank you for your comments. I have addressed the above suggestions by making appropriate changes except the test cases, which I have not included all yet. I will put them in my future commits. 

Yes, we do need a real switch in the environment for the test cases.
